pyang/plugins/name.py

```diff
@@ -21,7 +33,12 @@ def emit(self, ctx, modules, fd):
 def emit_name(ctx, modules, fd):
     for module in modules:
         bstr = ""
+        rstr = ""
+        if ctx.opts.print_revision:
+            r = module.search_one('revision')
```

Comment:
This will just print the first revision statement found, not necessarily the current.  Note that the strict ordering  of revision statements is just a SHOULD.   However, "module" has an attribute "i_latest_revision", which you can print, if it is not None.